File: LeetCode/[M] - 443. String Compression/solution.py
class Solution:
    def compress(self, chars: list[str]) -> int:
        p_symb, p_cnt = 0, 0
        res = 0
        while p_cnt != len(chars):
            ch_s, ch_cnt = chars[p_symb], chars[p_cnt]
            if ch_s == ch_cnt:
                p_cnt += 1
                res += 1
            else:
                if res < 2:
                    p_symb += 1
                    chars[p_symb] = chars[p_cnt]
                else:
                    p_symb += 1
                    for char in str(res):
                        chars[p_symb] = char
                        p_symb += 1
                    chars[p_symb] = chars[p_cnt]
                res = 0
        if res < 2:
            return p_symb + 1
        p_symb += 1
        for char in str(res):
            chars[p_symb] = char
            p_symb += 1
        return p_symb

File: LeetCode/[M] - 443. String Compression/solution.go
package main

import "strconv"

func compress(chars []byte) int {
	pSymb, pCnt := 0, 0
	cnt := 0
	for pCnt != len(chars) {
		if chars[pSymb] == chars[pCnt] {
			cnt += 1
			pCnt += 1
		} else {
			if cnt < 2 {
				pSymb += 1
				chars[pSymb] = chars[pCnt]
			} else {
				pSymb += 1
				for _, char := range strconv.Itoa(cnt) {
					chars[pSymb] = byte(char)
					pSymb += 1
				}
				chars[pSymb] = chars[pCnt]
			}
			cnt = 0
		}
	}
	if cnt < 2 {
		return pSymb + 1
	}
	pSymb += 1
	for _, char := range strconv.Itoa(cnt) {
		chars[pSymb] = byte(char)
		pSymb += 1
	}
	return pSymb
}
